app/assets/javascripts/listings/ListingDataService.js.coffee

```diff
@@ -0,0 +1,434 @@
+    labelMap =
+      "#{ListingConstantsService.RESERVED_TYPES.SENIOR}":
+        building: 'Senior'
+        eligibility: 'Seniors'
+        reservedFor: "seniors #{Service.seniorMinimumAge(listing)}"
+        reservedForWhoAre: "seniors #{Service.seniorMinimumAge(listing)}"
+        unitDescription: "seniors #{Service.seniorMinimumAge(listing)}"
+      "#{ListingConstantsService.RESERVED_TYPES.VETERAN}":
+        building: 'Veterans'
+        eligibility: 'Veterans'
+        reservedFor: 'veterans'
+        reservedForWhoAre: 'veterans'
+        unitDescription: 'veterans of the U.S. Armed Forces'
+      "#{ListingConstantsService.RESERVED_TYPES.DISABLED}":
+        building: 'Developmental Disability'
+        eligibility: 'People with developmental disabilities'
+        reservedFor: 'people with developmental disabilities'
+        reservedForWhoAre: 'developmentally disabled'
+        unitDescription: 'people with developmental disabilities'
+
+    return type unless labelMap[type]
+    return labelMap[type][modifier]
+
+  # used by My Applications -- when you load an application we also parse the attached listing data
+  Service.loadListing = (listing) ->
+    return if Service.listing && Service.listing.Id == listing.Id
+    # TODO: won't be needed if we ever consolidate Listing_Lottery_Preferences and /preferences API
+    listing.preferences = _.map listing.Listing_Lottery_Preferences, (lotteryPref) ->
+      {
+        listingPreferenceID: lotteryPref.Id
+        preferenceName: lotteryPref.Lottery_Preference.Name
+      }
+    angular.copy(listing, Service.listing)
+
+  Service.seniorMinimumAge = (listing) ->
+    if listing.Reserved_community_minimum_age
+      "#{listing.Reserved_community_minimum_age}+"
+    else
+      ''
+
+  Service.formattedAddress = (listing, type='Building', display='full') ->
+    street = "#{type}_Street_Address"
+    zip = "#{type}_Postal_Code"
+    if type == 'Leasing_Agent'
+      street = "#{type}_Street"
+      zip = "#{type}_Zip"
+    else if type == 'Building'
+      zip = "#{type}_Zip_Code"
+
+    # If Street address is undefined, then return false for display and google map lookup
+    if listing[street] == undefined
+      return
+    # If other fields are undefined, proceed, with special string formatting
+    if listing[street] != undefined
+      Street_Address = listing[street] + ', '
+    else
+      Street_Address = ''
+    if listing["#{type}_City"] != undefined
+      City = listing["#{type}_City"]
+    else
+      City = ''
+    if listing["#{type}_State"] != undefined
+      State = listing["#{type}_State"]
+    else
+      State = ''
+    if listing[zip] != undefined
+      Zip_Code = listing[zip]
+    else
+      Zip_Code = ''
+
+    if display == 'street'
+      return "#{Street_Address}"
+    else if display == 'city-state-zip'
+      return "#{City} #{State}, #{Zip_Code}"
+    else
+      "#{Street_Address}#{City} #{State}, #{Zip_Code}"
+
+  Service.getListingDownloadURLs = ->
```

Comment:
it would be really great to rename this to something a bit clearer. like getApplicationDownloadUrl or getPaperApplicationDownload url. Thoughts?

We can save it until we add the paper applications for ownership listings if that's easier?

Reply:
yes agree, the name is misleading, also agree let's leave it for that story where we deal with paper apps for ownership. i've put a note to do that in the relevant story: https://www.pivotaltracker.com/story/show/162975138